### PR TITLE
Cleanup setup.py

### PR DESCRIPTION
**Pull request by tamird**
Roughly a combination of #131 and #161 and closes both. @haberman

### CODE REVIEW

**Comment by haberman** [comment]
In this project we use 2-space indentation for Python. Would you mind updating the files to reflect this?

**Comment by tamird** [comment]
@haberman done

**Comment by tamird** [comment]
@haberman do you use 4-space indent for continuations?

File: python/setup.py [hunk 1]
```diff
@@ -145,61 +151,52 @@ def run(self):
...
+      )
+    )
     os.environ['PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION'] = 'cpp'
 
-  setup(name = 'protobuf',
-        version = GetVersion(),
-        packages = [ 'google' ],
-        namespace_packages = [ 'google' ],
-        test_suite = 'google.protobuf.internal',
-        # Must list modules explicitly so that we don't install tests.
```
Comment (haberman) [inline python/setup.py]: Doesn't this comment still apply? Seems like we were listing individual modules for a reason.

Comment (tamird) [inline python/setup.py]: I believe that's covered by `exclude google/protobuf/internal/*_test.py` in MANIFEST.in

**Comment by tamird** [comment]
@haberman added a few commits to get tests passing under 2.6. PTAL?

**Comment by tamird** [comment]
@haberman this is green; mind having a once-over?

**Comment by haberman** [comment]
Any chance this works with Python 2.5?

Also, ideally we'd be able to actually test it under Python 2.5/2.6 using Travis. It's hard to keep things compatible if we are not regularly testing that compatibility. Hopefully this could be added to the Travis build without too much trouble?

**Comment by tamird** [comment]
@haberman I did a lot of googling this morning to work out how to get travis to build a matrix of languages and versions of those languages (currently this project tests with multiple JDKs), and landed on this (still open) issue: https://github.com/travis-ci/travis-ci/issues/2877, so it seems that this isn't possible without quite a bit of work, right now.

I have another commit locally that makes the tests pass under 2.5 - I'll push it now, but it's pretty ugly and way more invasive. Also, the cpp implementation does compile with 2.5, because it uses `PyBytes_FromStringAndSize` which returns a [`bytearray`](https://docs.python.org/2/library/functions.html#bytearray), which was added in 2.6.

**Comment by tamird** [comment]
@haberman what are you thinking?

**Comment by haberman** [comment]
Sorry for the delay -- I'm trying to nail down whether it's feasible to drop 2.5 support right now by freezing the runtime used by AppEngine's 2.5 runtime.

If we do that, then we can set the minimum supported version to either 2.6 or 2.7 and avoid a lot of pain.

**Comment by tamird** [comment]
Makes sense! Let me know.

**Comment by haberman** [comment]
Ok, we have a green-light to remove Python 2.5 compatibility from this repo!

**Comment by haberman** [comment]
I am inclined to say that we drop support for Python < 2.7. Thoughts?

**Comment by tamird** [comment]
Yep, I'm in agreement. Want me to update this PR to do that?

File: python/setup.py [hunk 2]
```diff
@@ -145,61 +150,49 @@ def run(self):
...
+      name='protobuf',
+      version=GetVersion(),
+      description='Protocol Buffers',
+      long_description="Protocol Buffers are Google's data interchange format",
+      url='https://developers.google.com/protocol-buffers/',
+      maintainer='protobuf@googlegroups.com',
+      maintainer_email='protobuf@googlegroups.com',
+      license='New BSD License',
+      classifiers=[
+          'Programming Language :: Python :: 2.7',
```
Comment (tamird) [inline python/setup.py]: this suffices to support only 2.7, right?

**Comment by haberman** [comment]
Sure. If we don't have to worry about Python 2.6, this PR gets a lot shorter/simpler, right?

**Comment by tamird** [comment]
Yeah, the last two commits are gone

**Comment by tamird** [comment]
Green

**Comment by haberman** [comment]
Thanks so much for this work and for your patience.

If you are able, since you are modifying this part of the code already, could you possibly take a look at this bug and see if it would make sense to fix as part of this PR also?

https://github.com/google/protobuf/issues/333

I'm not too familiar with Python setuptools, so I don't know the right answer here. The main symptom is that when I tried to upload protobuf-python v3.0.0-alpha-2 to PyPI using the command `python setup.py sdist bdist_egg upload`, the resulting package could not install successfully.

If you'd rather punt on this, that's ok too. I just wanted to mention it now since it seems like a prime time to address it.

**Comment by tamird** [comment]
I don't know the solution either; let's punt and maybe @tseaver can contribute a fix on top of this PR.

**Comment by tamird** [comment]
Would you mind closing #161?

**Pull request merged**